src/_Classics_/knuth-morris-pratt/index.js

```diff
@@ -0,0 +1,100 @@
+
+/* *
+* The time complexity of KMP algorithm is O(n) in the worst case
+* Example use case: Pattern = AABCAB Text = AAABACABAABCABAABCA
+* LPSArray = [ 0, 0, 1, 2, 3, 0 ]
```

Comment:
It should be LPSArray = [ 0, 1, 0, 0, 1, 0 ]

Reply:
Apologies, code results in this LPS: [ 0, 1, 0, 0, 1, 0 ]. I was using different inputs and forgot to swap out the LPS output for the new values I included in this example. Will update the PR with new test data.

Will update the example to show for: KMPSearch("ABCABCACA","AAABCBAABCABCACACABBCA");

Where: 
P = ABCABCACA
T = AAABCBAABCABCACACABBCA
lpsArray: [ 0, 0, 0, 1, 2, 3, 4, 0, 1 ]
Pattern Found at index 7

I feel this is a more useful example than the one I included above.

Reply:
PR has been updated with the new example, cheers.